EKF/control.cpp

```diff
@@ -773,10 +773,16 @@ void Ekf::controlHeightFusion()
 		// we have just switched to using range finder, calculate height sensor offset such that current
 		// measurment matches our current height estimate
 		if (_control_status_prev.flags.rng_hgt != _control_status.flags.rng_hgt) {
-			if (_terrain_initialised) {
+			// use the parameter rng_gnd_clearance if on ground to avoid a noisy offset initialization (e.g. sonar)
+			if (_control_status.flags.in_air && _terrain_initialised) {
+
 				_hgt_sensor_offset = _terrain_vpos;
```

Comment:
I think we should also constrain _terrain_vpos so that it is always a minimum of _params.rng_gnd_clearance larger than _state.pos(2).